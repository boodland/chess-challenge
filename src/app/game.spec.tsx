import { render } from '@testing-library/react';

import Game from './game';
import { ReactNode } from 'react';

const gameProviderMockedName = 'MockedGameProvider';
vi.mock('./game-context', () => ({
  GameProvider: ({ children }: { children: ReactNode }) => (
    <div>
      <div>{gameProviderMockedName}</div>
      <div>{children}</div>
    </div>
  ),
}));

const boardMockedName = 'MockedBoard';
vi.mock('./board', () => ({
  default: () => <div>{boardMockedName}</div>,
}));

const gameActionsMockedName = 'MockedGameActions';
vi.mock('./game-actions', () => ({
  default: () => <div>{gameActionsMockedName}</div>,
}));

describe('Game', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Game />);
    expect(baseElement).toBeTruthy();
  });

  it('should have the correct styles', () => {
    const { container } = render(<Game />);
    expect(container).toMatchSnapshot();
  });

  it('should display GameProvider component', () => {
    const { getByText } = render(<Game />);
    expect(getByText(gameProviderMockedName)).toBeTruthy();
  });

  it('should display Board component', () => {
    const { getByText } = render(<Game />);
    expect(getByText(boardMockedName)).toBeTruthy();
  });

  it('should display GameActions component', () => {
    const { getByText } = render(<Game />);
    expect(getByText(gameActionsMockedName)).toBeTruthy();
  });
});
