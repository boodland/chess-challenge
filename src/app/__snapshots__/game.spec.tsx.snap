// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game > should have the correct styles 1`] = `
<div>
  <div>
    <div>
      MockedGameProvider
    </div>
    <div>
      <div
        class="_game_b500ea"
      >
        <div>
          <div>
            MockedBoard
          </div>
        </div>
        <div
          class="_right-panel_b500ea"
        >
          <div>
            MockedGameActions
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
